### ISSUE TITLE
No errors reported when type-hinting  a class variable as Final[ClassVar[int]] 

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#2120]
**Describe the bug**
No errors are produced when checking a class variable that has been typed-hinted with something like `Final[ClassVar[int]]`

**To Reproduce**

```py3
#pyright: strict

from typing import ClassVar
from typing_extensions import Final

class MyClass:
    # no pyright errors, but fails during runtime with:
    # TypeError: typing.ClassVar[int] is not valid as type argument
    MY_CONST: Final[ClassVar[int]] = 123 
```

**Expected behavior**
Pyright should flag the type-hinting of `MY_CONST` as an error. As per [PEP591](https://www.python.org/dev/peps/pep-0591/): "Type checkers should infer a final attribute that is initialized in a class body as being a class variable. Variables should not be annotated with both ClassVar and Final."

**VS Code extension or command-line**
cli version 1.1.158
pyright vscode exgtension version 1.1.157

**Comment by erictraut** [comment]
Thanks for the suggestion. This will be included in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
